src/foam/lib/json/Outputter.java

```diff
@@ -265,6 +269,10 @@ protected Boolean maybeOutputProperty(FObject fo, PropertyInfo prop, boolean inc
     return true;
   }
 
+  protected  boolean propertyPredicate( FObject fo, PropertyInfo prop) {
```

Comment:
This method should be in an interface: PropertyPredicate. Then this class can use the Strategy pattern, not the Template pattern, as you've done. This would mean that you could reuse the PermissionedPropertyPredicate for the XMLOutputter and other Outputters as well. Also, we could then get rid of the mode_ variable and make Network and Storage be constant predicates.

Reply:
> This method should be in an interface: PropertyPredicate. Then this class can use the Strategy pattern, not the Template pattern, as you've done. This would mean that you could reuse the PermissionedPropertyPredicate for the XMLOutputter and other Outputters as well. Also, we could then get rid of the mode_ variable and make Network and Storage be constant predicates.

Why Strategy pattern? Isn't PermissionedPropertyPredicate identical for JSON outputter, XML outputter, etc?

Reply:
Yes, PermissionedPropertyPredicate should be identical for all outputters. But you don't have that interface here, it's just a template method that you override by subclassing. You don't have it as a standalone reusable object.